### ISSUE TITLE
`prefer-arrow-callback` with `mocha/no-mocha-arrows`

### ISSUE THREAD

**Issue by Bamieh** [eslint/eslint#9255]
**Tell us about your environment**
I am running tests in mocha that i want to eslint out!

* **ESLint Version:** ^4.6.1
* **Node Version:** v8.1.3
* **npm Version:** 5.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: [
    'airbnb',
    'prettier',
  ],
  plugins: [
    'prettier',
    'mocha',
  ],
  globals: {
    chai: true,
    logger: true,
  },
  env: {
    node: true,
    mocha: true,
    browser: false,
  },
  rules: {
    'no-console': 'error',
    'prefer-arrow-callback': 2,
    'mocha/no-mocha-arrows': 'error',
    'prettier/prettier': ['error', {
      singleQuote: true,
      bracketSpacing: false,
      trailingComma: 'all',
    }],
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
describe("my suite", function() {
});
```

**What did you expect to happen?**
The above line should not error out the linter. Since i am using mocha, and mocha prefers anonymous function, not an arrow function.

Currently it errors out because of this rule:

```
rules: {
    'prefer-arrow-callback': 2,
  ...
```

I am expecting the arrow functions to be preferred everywhere except inside a case function or a suite function.

The prefer arrow-callback errors out, but it shouldn't, probably it needs a special configuration.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ x:xx  error    Unexpected function expression     prefer-arrow-callback
```

**Comment by platinumazure** [comment]
You could use [disable comments](https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments) to disable reports, although that wouldn't be a good solution in general.

Maybe we could enhance the rule to avoid reporting on function expressions that fit a user-specified selector? But I'm not sure the benefit is worth the maintenance burden for a core rule, at this point.

I think your best option is to write an issue for eslint-plugin-mocha asking them to create their own version of prefer-arrow-callback which does not report for Mocha test callbacks.

**Comment by Bamieh** [comment]
Makes sense

**Comment by paulmelnikow** [comment]
@Bamieh I'm just curious, did you open an issue on eslint-plugin-mocha? I would like to use this functionality as well.

**Comment by Bamieh** [comment]
@paulmelnikow no i did not

**Comment by paulmelnikow** [comment]
Opened lo1tuma/eslint-plugin-mocha#141.
